docs/book/user-guide/best-practices/quick-wins.md

```diff
@@ -321,7 +341,163 @@ zenml artifact-store register my_store \
+    full_data = load_your_dataset()
+    
+    # Only use a small fraction during smoke testing
+    if sample_fraction < 1.0:
+        sampled_data = sample_dataset(full_data, sample_fraction)
+        print(f"SMOKE TEST MODE: Using {sample_fraction*100}% of data")
+        return sampled_data
+    
+    return full_data
+
+# 3. Run pipeline with the local Docker orchestrator
+training_pipeline(sample_fraction=0.01)
+```
+
+**When to switch back to cloud**
+```bash
+# When your smoke tests pass, switch back to your cloud stack
+zenml stack set production_stack  # Your cloud-based stack
+
+# Run the same pipeline with full data
+training_pipeline(sample_fraction=1.0)  # Use full dataset
+```
+
+**Key benefits**
+* **Fast feedback cycles** - Get results in minutes instead of hours
+* **Cost savings** - Test on your local machine instead of paying for cloud resources
+* **Simplified debugging** - Easier access to logs and containers
+* **Consistent environments** - Same Docker containerization as production
+* **Reduced friction** - No cloud provisioning delays or permission issues during development
+
+**Best practices**
+* Create a small representative dataset for smoke testing
+* Use configuration parameters to enable smoke-test mode
+* Keep dependencies identical between smoke tests and production
+* Run the exact same pipeline code locally and in the cloud
+* Store sample data in version control for reliable testing
+* Use `prints` or logging to clearly indicate when running in smoke-test mode
+
+This approach works best when you design your pipelines to be configurable from the start, allowing them to run with reduced data size, shorter training cycles, or simplified processing steps during development.
+
+Learn more: [Local Docker Orchestrator](https://docs.zenml.io/stack-components/orchestrators/local-docker)
+
+## 9 Organize with tags
+
+**Why** -- add flexible, searchable labels to your ML assets that bring order to chaos as your project grows. Tags provide a lightweight organizational system that helps you filter pipelines, artifacts, and models by domain, status, version, or any custom category—making it easy to find what you're looking for in seconds.
+
+```python
+from zenml import pipeline, step, add_tags, Tag
+
+# 1. Tag your pipelines with meaningful categories
+@pipeline(tags=["fraud-detection", "training", "financial"])
+def training_pipeline():
+    # Your pipeline steps
+    preprocess_step(...)
+    train_step(...)
+    evaluate_step(...)
+
+# 2. Create "exclusive" tags for state management
+@pipeline(tags=[
+    Tag(name="production", exclusive=True),  # Only one pipeline can be "production"
+    "financial"
+])
+def production_pipeline():
+    pass
+
+# 3. Tag artifacts programmatically from within steps
+@step
+def evaluate_step():
+    # Your evaluation code here
+    accuracy = 0.95
+    
+    # Tag based on performance
+    if accuracy > 0.9:
+        add_tags(tags=["high-accuracy"], infer_artifact=True)
```

Comment:
hmm what is infer_artifact?

Reply:
From the docstring: "Flag deciding whether the artifact version should be
            inferred from the step context."